### DISCUSSION TITLE
UI vs Daemon

### DISCUSSION THREAD

**Discussion by baarkerlounger** [quexten/goldwarden#47] [Q&A] [answered]
When installed via flatpak does the daemon always run in the background? Or only when the app has been launched (e.g. via desktop icon or flatpak run command).

**Comment by quexten** [comment, marked as answer]
Thanks for the question. If you just install the Flatpak, the daemon is started inside Flatpak, and the UI is spawned as a separate process. If you close the UI, only the UI process exits, the daemon keeps running. In gnome this is shown as a background activity, I'm guessing KDE and other DE's have something similar
![image](https://github.com/quexten/goldwarden/assets/11866552/17479a19-ab47-4cb7-a336-486c93f5cd93)

If you want to stop this, in Gnome you would just click the "x" in the background apps. If you want to disable the behavior, you can take away the background activity permission in Gnome's settings.

Additionally, the backround portal is used, to auto-start the Flatpak on the next boot, so that auto-type and ssh work without you manually starting the Flatpak through the desktop file/your DE's launcher every time.

*IF* you have configured an external daemon to auto-start via systemd, and listen on the socket path of the Flatpak, then the Flatpak uses the external daemon and does not start it's own daemon.

**Comment by baarkerlounger** [comment]
Thanks understood. It might be nice to be able to configure from the UI whether the daemon continues to run in the background or stops with the UI and whether or not it's autostarted. (That might also help to make the default behaviour a bit more obvious).

**Reply by quexten** [reply]
Agreed, added it as an enhancement issue!
